docker-compose.yml

```
@@ -3,11 +3,9 @@ version: "3.8"
 services:
   hugo:
     build:
-      # To build from github, you need to install docker-compose version=1.27.1 (Does not work with the last versions)
-      context: https://github.com/gohugoio/hugo.git
+      context: infra/hugo
```

The build context is automatically pushed to the docker build command: https://docs.docker.com/engine/reference/commandline/build/#git-repositories
If you want to target a specific version, you should follow the documentation in order to point to the wanted version, for example.
```suggestion
      context: [infra/hugov](https://github.com/gohugoio/hugo.git#0.96.0
```
Otherwise, I  see no reason to have a submodule that is not mandatory for building the app :thinking:

---

When i try to run docker-compose up with github context, I encounter a cryptic error about `OPENSSL_1_1_1`. The only hint i Found is `# To build from github, you need to install docker-compose version=1.27.1 (Does not work with the last versions)` which means downgrading my system to use this repository.

Submodules was the easy solution I found to solve my problem. If not satisfying, can you help to find a better one ?

Ps. From what i found in [github actions](https://github.com/PrestaShop/devdocs-site/blob/main/.github/workflows/build.yml), current used version is 0.85.0

---

```bash
 HUGO_SERVER_PORT=3031 docker-compose build
Building hugo
Sending build context to Docker daemon  9.626MB
Step 1/18 : FROM golang:1.18-alpine AS build
1.18-alpine: Pulling from library/golang
40e059520d19: Pull complete 
3b494d572400: Pull complete 
623a0a71bae0: Pull complete 
b9e55b4f0af9: Pull complete 
1eb0e4976eb6: Pull complete 
Digest: sha256:3afd220509acf9866e91932a3a41bf341b8bada82107ef3ecce3422826b98064
Status: Downloaded newer image for golang:1.18-alpine
 ---> 99760761bfe0
Step 2/18 : ARG HUGO_BUILD_TAGS
 ---> Running in 1b081ce152b9
Removing intermediate container 1b081ce152b9
 ---> 790b0e185214
Step 3/18 : ARG CGO=1
 ---> Running in 872c5904dd94
Removing intermediate container 872c5904dd94
 ---> 62dcf599b871
Step 4/18 : ENV CGO_ENABLED=${CGO}
 ---> Running in 85b678ca6159
Removing intermediate container 85b678ca6159
 ---> c576fd6bdb71
Step 5/18 : ENV GOOS=linux
 ---> Running in 5a760eb71830
Removing intermediate container 5a760eb71830
 ---> fc45c9924927
Step 6/18 : ENV GO111MODULE=on
 ---> Running in 27c476788b88
Removing intermediate container 27c476788b88
 ---> 7fc0a99b0334
Step 7/18 : WORKDIR /go/src/github.com/gohugoio/hugo
 ---> Running in f6ed24bb17c8
Removing intermediate container f6ed24bb17c8
 ---> e8d0350be3c8
Step 8/18 : COPY . /go/src/github.com/gohugoio/hugo/
 ---> 56bb9ca5ebd4
Step 9/18 : RUN apk update &&     apk add --no-cache gcc g++ musl-dev git &&     go install github.com/magefile/mage
 ---> Running in b85f8d029682
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/community/x86_64/APKINDEX.tar.gz
v3.15.3-21-g2745de7e1b [https://dl-cdn.alpinelinux.org/alpine/v3.15/main]
v3.15.3-17-ge0ce01e898 [https://dl-cdn.alpinelinux.org/alpine/v3.15/community]
OK: 15856 distinct packages available
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/community/x86_64/APKINDEX.tar.gz
(1/20) Installing libgcc (10.3.1_git20211027-r0)
(2/20) Installing libstdc++ (10.3.1_git20211027-r0)
(3/20) Installing binutils (2.37-r3)
(4/20) Installing libgomp (10.3.1_git20211027-r0)
(5/20) Installing libatomic (10.3.1_git20211027-r0)
(6/20) Installing libgphobos (10.3.1_git20211027-r0)
(7/20) Installing gmp (6.2.1-r1)
(8/20) Installing isl22 (0.22-r0)
(9/20) Installing mpfr4 (4.1.0-r0)
(10/20) Installing mpc1 (1.2.1-r0)
(11/20) Installing gcc (10.3.1_git20211027-r0)
(12/20) Installing musl-dev (1.2.2-r7)
(13/20) Installing libc-dev (0.7.2-r3)
(14/20) Installing g++ (10.3.1_git20211027-r0)
(15/20) Installing brotli-libs (1.0.9-r5)
(16/20) Installing nghttp2-libs (1.46.0-r0)
(17/20) Installing libcurl (7.80.0-r0)
(18/20) Installing expat (2.4.7-r0)
(19/20) Installing pcre2 (10.39-r0)
(20/20) Installing git (2.34.1-r0)
Executing busybox-1.34.1-r4.trigger
OK: 202 MiB in 35 packages
go: downloading github.com/magefile/mage v1.12.1
Removing intermediate container b85f8d029682
 ---> 20fe594c112b
Step 10/18 : RUN mage hugo && mage install
 ---> Running in adb7a1e5c407
fatal: not a git repository (or any of the parent directories): .git
go: downloading github.com/bep/golibsass v1.0.0

fatal: not a git repository (or any of the parent directories): .git
Removing intermediate container adb7a1e5c407
 ---> ed8f276b6666
Step 11/18 : FROM alpine:3.12
3.12: Pulling from library/alpine
9c4930dab762: Pull complete 
Digest: sha256:e42c1b8e03bdaa471deb198d981b52ac421441975195a20004aea624890f8b1c
Status: Downloaded newer image for alpine:3.12
 ---> d7db87feda9d
Step 12/18 : COPY --from=build /go/bin/hugo /usr/bin/hugo
 ---> c299cfa839a2
Step 13/18 : RUN apk update &&     apk add --no-cache ca-certificates libc6-compat libstdc++ git
 ---> Running in f67d9bd9b1e5
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/x86_64/APKINDEX.tar.gz
v3.12.11-4-g41c9e389fd [http://dl-cdn.alpinelinux.org/alpine/v3.12/main]
v3.12.10-17-g0c8862530f [http://dl-cdn.alpinelinux.org/alpine/v3.12/community]
OK: 12765 distinct packages available
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/x86_64/APKINDEX.tar.gz
(1/9) Installing ca-certificates (20211220-r0)
(2/9) Installing nghttp2-libs (1.41.0-r0)
(3/9) Installing libcurl (7.79.1-r0)
(4/9) Installing expat (2.2.10-r4)
(5/9) Installing pcre2 (10.35-r0)
(6/9) Installing git (2.26.3-r1)
(7/9) Installing libc6-compat (1.1.24-r10)
(8/9) Installing libgcc (9.3.0-r2)
(9/9) Installing libstdc++ (9.3.0-r2)
Executing busybox-1.31.1-r21.trigger
Executing ca-certificates-20211220-r0.trigger
OK: 24 MiB in 23 packages
Removing intermediate container f67d9bd9b1e5
 ---> 45d18adaf74e
Step 14/18 : VOLUME /site
 ---> Running in 1dea927de40d
Removing intermediate container 1dea927de40d
 ---> 2810142e2224
Step 15/18 : WORKDIR /site
 ---> Running in 3dcd4ffaa513
Removing intermediate container 3dcd4ffaa513
 ---> 31dcce39a46e
Step 16/18 : EXPOSE 1313
 ---> Running in 269a27fa9a0d
Removing intermediate container 269a27fa9a0d
 ---> d78c208d7ce0
Step 17/18 : ENTRYPOINT ["hugo"]
 ---> Running in 84222b60d1e0
Removing intermediate container 84222b60d1e0
 ---> b0f1bcde8791
Step 18/18 : CMD ["--help"]
 ---> Running in 30b08aca5463
Removing intermediate container 30b08aca5463
 ---> 02c4b8cc3107
Successfully built 02c4b8cc3107
Successfully tagged devdocs-site_hugo:latest
[3.8.3] got@got:~/projects/prestashop/devdocs-site[main !]$ 
[3.8.3] got@got:~/projects/prestashop/devdocs-site[main !]$ HUGO_SERVER_PORT=3031 docker-compose up
Creating network "devdocs-site_default" with the default driver
Creating devdocs-site_hugo_1 ... done
Attaching to devdocs-site_hugo_1
hugo_1  | Start building sites … 
hugo_1  | hugo v0.97.0-DEV+extended linux/amd64 BuildDate=unknown VendorInfo=mage
hugo_1  | 
hugo_1  |                    |  EN   
hugo_1  | -------------------+-------
hugo_1  |   Pages            | 1863  
hugo_1  |   Paginator pages  |    0  
hugo_1  |   Non-page files   |  402  
hugo_1  |   Static files     |   66  
hugo_1  |   Processed images |    0  
hugo_1  |   Aliases          |  141  
hugo_1  |   Sitemaps         |    1  
hugo_1  |   Cleaned          |    0  
hugo_1  | 
hugo_1  | Built in 11265 ms
hugo_1  | Watching for changes in /site/src/{content,layouts,static,themes}
hugo_1  | Watching for config changes in /site/src/config.yml
hugo_1  | Environment: "development"
hugo_1  | Serving pages from memory
hugo_1  | Running in Fast Render Mode. For full rebuilds on change: hugo server --disableFastRender
hugo_1  | Web Server is available at http://localhost:3031/ (bind address 0.0.0.0)
hugo_1  | Press Ctrl+C to stop
```

```

---

I check, everything seems to be working well inside the doc website

---

Maybe the real change that need to be done is to update the compatibility with latest hugo version to be sure everything is ok :)

---

>When i try to run docker-compose up with github context, I encounter a cryptic error about OPENSSL_1_1_1.

Ab out this error, you should install `docker-compose`  with pip instead of the curl command. It will take in consideration what you have on your system, the curl command is build based on someone else computer

---

I redo my tests with a clean install, and everything is ok on my own :thinking: 
(I use the `#v0.85.0` version)

---

@PierreRambaud I'm a noob in docker so I cant review this PR. Are you Ok with submitted content?

---

As I said above, I see no reason to add a submodule as the docker-compose already includes it

---

Sorry I didn't test @PierreRambaud solution yet (I have to reinstall a part of my system to test it and I am on something else ^^')

I'll try to come back in few days.